src/ProcessManager.php

```diff
@@ -724,14 +729,15 @@ protected function bootstrapFailed($port)
      */
     protected function checkChangedFiles($restartSlaves = true)
     {
-        if ($this->inReload) {
+        if ($this->inChangesDetectionCycle) {
             return false;
         }
 
+        $this->inChangesDetectionCycle = true;
+
         clearstatcache();
 
-        $reload = false;
-        $filePath = '';
+        $modifiedFilesLog = [];
```

Comment:
Is this needed instead of logging right away?

Reply:
cons: increasing memory footprint, pros: decreasing IO operations.

should I rewrite it to in-place logging?

Reply:
> should I rewrite it to in-place logging?

Since this only happens in debug mode and we're breaking on first mismatch (see below) I don't think it needs the buffering.